openslides/__main__.py

```diff
@@ -86,283 +74,85 @@ def parse_args():
+        'createsettings',
+        help='Create the settings file.')
+    subcommand_createsettings.set_defaults(callback=createsettings)
+    subcommand_createsettings.add_argument(
+        '--settings_path',
+        action='store',
+        default=None,
+        help='The used settings file. The file is created, even if it exists.')
+    subcommand_createsettings.add_argument(
+        '--development',
+        action='store_true',
+        help='Command for development purposes.')
 
-def syncdb(settings, args):
-    """
-    Run syncdb to create or update the database.
-    """
-    ensure_settings(settings, args)
-    db_file = get_database_path_from_settings()
-    if db_file is not None:
-        db_dir = os.path.dirname(db_file)
-        if not os.path.exists(db_dir):
-            os.makedirs(db_dir)
-        if not os.path.exists(db_file):
-            print('Clearing old search index...')
-            execute_from_command_line(["", "clear_index", "--noinput"])
-    execute_from_command_line(["", "syncdb", "--noinput"])
-    if args.language:
-        translate_customizable_strings(args.language)
-    return 0
+    return parser
 
 
-def createsuperuser(settings, args):
+def start(args):
     """
-    Creates or resets the admin user. Returns 0 to show success.
+    Starts OpenSlides: Runs migrations and runs runserver.
     """
-    ensure_settings(settings, args)
-    # can't be imported in global scope as it already requires
-    # the settings module during import
-    from openslides.users.api import create_or_reset_admin_user
-    if create_or_reset_admin_user():
-        print('Admin user successfully created.')
-    else:
-        print('Admin user successfully reset.')
-    return 0
-
+    settings_path = args.settings_path
+    development = is_development()
 
-def backupdb(settings, args):
-    """
-    Stores a backup copy of the SQlite3 database. Returns 0 on success, else 1.
-    """
-    ensure_settings(settings, args)
+    if settings_path is None:
+        if development:
+            settings_path = get_development_settings_path()
+        else:
+            settings_path = get_default_settings_path()
 
-    from django.db import connection, transaction
+    # Write settings if it does not exists.
+    if not os.path.isfile(settings_path):
+        createsettings(args)
 
-    @transaction.atomic
-    def do_backup(src_path, dest_path):
-        # perform a simple file-copy backup of the database
-        # first we need a shared lock on the database, issuing a select()
-        # will do this for us
-        cursor = connection.cursor()
-        cursor.execute("SELECT count(*) from sqlite_master")
-        # now copy the file
-        try:
-            shutil.copy(src_path, dest_path)
-        except IOError:
-            raise Exception("Database backup failed.")
+    # Set the django setting module and run migrations
+    setup_django_settings_module(settings_path, development=development)
```

Comment:
The start command can not use the environment variable, because it wants to create the setting, if it does not exist. But where do you create a settings file, when the environment variable is "example.settings" ?

I made this clearer in the help message and in some comments. I also changed the function setup_django_settings_module(). Now, it does not need the argument "keep_environment_variable".